bundle/Controller/WebsiteToolbarController.php

```diff
@@ -109,6 +124,36 @@ public function websiteToolbarAction($locationId, Request $request)
         return $response;
     }
 
+    /**
+     * Build the response so that depending on settings it's cacheable.
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    protected function buildResponse()
+    {
+        $request = $this->getRequest();
+        $response = new Response();
+        if ($this->viewCache === true) {
+            $response->setPublic();
+
+            if ($this->ttlCache === true) {
+                $response->setSharedMaxAge(
+                    $this->defaultTtl
+                );
+            }
+
+            // Make the response vary against Cookie header ensures that an HTTP
+            // reverse proxy caches the different possible variations of the
+            // response as it can depend on user role for instance. X-User-Hash cannot
+            // be used since the website toolbar can have Owner( Self ) Policy Limitation.
+            if ($request->headers->has('Cookie')) {
+                $response->setVary('Cookie');
```

Comment:
wondering if we should always set this even if there are no cookie, as otherwise the non varying cache will be set in varnish and used also by requests with cookies afaik.

Reply:
I'll check that to be sure and change it if needed.

Reply:
On further taught: In practice it provably won't happen _if_ eZ puts vary by user hash in request as well.

Reply:
@andrerom from my tests it looks like eZ indeed puts vary by user hash in the request. But otherwise, vary by cookie would be needed. So I'll change that fragment to always set it, just to be on the safe side.